modules/ingest-user-agent/src/test/java/org/elasticsearch/ingest/useragent/DeviceTypeParserTests.java

```diff
@@ -0,0 +1,217 @@
+                        testDevice.put("os", map.get("os"));
+                        testDevice.put("browser", map.get("browser"));
+                        testDevice.put("device", map.get("device"));
+                        testDevices.add(testDevice);
+
+                    }
+                }
+            }
+        }
+
+        return testDevices;
+    }
+
+    private static VersionedName getVersionName(String name){
+        return new VersionedName(name, null, null, null, null);
+    }
+
+    @BeforeClass
+    public static void setupDeviceParser() throws IOException {
+        InputStream deviceTypeRegexStream = UserAgentProcessor.class.getResourceAsStream("/device_type_regexes.yml");
+
+        assertNotNull(deviceTypeRegexStream);
+        assertNotNull(deviceTypeRegexStream);
+
+        deviceTypeParser = new DeviceTypeParser();
+        deviceTypeParser.init(deviceTypeRegexStream);
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testMacDesktop() throws Exception {
+        VersionedName os = getVersionName("Mac OS X");
+
+        VersionedName userAgent = getVersionName("Chrome");
+
+        String deviceType = deviceTypeParser.findDeviceType(userAgent, os, null);
+
+        assertThat(deviceType, is("Desktop"));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testAndroidMobile() throws Exception {
+
+        VersionedName os = getVersionName("iOS");
+
+        VersionedName userAgent = getVersionName("Safari");
+
+        String deviceType = deviceTypeParser.findDeviceType(userAgent, os, null);
+
+        assertThat(deviceType, is("Phone"));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testIPadTablet() throws Exception {
+
+        VersionedName os = getVersionName("iOS");
+
+        VersionedName userAgent = getVersionName("Safari");
+
+        VersionedName device = getVersionName("iPad");
+
+        String deviceType = deviceTypeParser.findDeviceType(userAgent, os, device);
+
+        assertThat(deviceType, is("Tablet"));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testWindowDesktop() throws Exception {
+
+        VersionedName os = getVersionName("Mac OS X");
+
+        VersionedName userAgent = getVersionName("Chrome");
+
+        String deviceType = deviceTypeParser.findDeviceType(userAgent, os, null);
+
+        assertThat(deviceType, is("Desktop"));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testRobotAgentString() throws Exception {
+
+        String deviceType = deviceTypeParser.findDeviceType(
+            "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:63.0.247) Gecko/20100101 Firefox/63.0.247 Site24x7", null, null, null);
+
+        assertThat(deviceType, is("Robot"));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testRobotDevices() throws Exception {
+
+        InputStream deviceTypeRegexStream = IngestUserAgentPlugin.class.getResourceAsStream("/robot-devices.yml");
```

Comment:
Can you help me understand the purpose of testing with device type patterns other than the ones defined in `device_type_regexes.yml`?